### DISCUSSION TITLE
Can I change my destination to become my source ?

### DISCUSSION THREAD

**Discussion by mehdi-wsc** [backube/volsync#206] [Q&A]
Hello gents,
I'm using volsync for my DR plan  (OpenShift clusters), I have a requirement that the destination cluster becomes the source in case of disaster , Can I achieve this with volsync using the same S3 bucket ? 

Regards,

**Comment by JohnStrunk** [comment]
Since you mention an S3 bucket, I assume you're using either the rclone or restic movers.

You can use the same bucket, but using the same path within that bucket could get you into trouble.

If the goal is just to not have a bunch of different buckets, then go ahead and use the same bucket w/ different paths. The data will remain separate, but you'll only have to manage one set of bucket parameters. To get different paths, take a look at [`rcloneDestPath`](https://volsync.readthedocs.io/en/latest/usage/rclone/index.html#additional-source-options).

If the goal is to re-use the data that was already in the bucket when you "reverse" the direction, that's more dangerous, and I'd advise against it.
Consider the following (assumes rclone):
- Starting condition: Site1 is primary, site2 is secondary, replicating via a single bucket/path
- Site1 fails
- Site2 is brought online by promoting the most recent snapshot
- Site2 is configured to sync to the bucket (as a new source)
- Site1 "recovers" enough that pods get scheduled and external network connectivity is available
- Site 1 would then resume syncing (also as a source) w/ the bucket, corrupting the data in the bucket
  - Bucket contents would be unusable until you get rid of the site1 source CR and give site2 a chance to fix the bucket contents. This would leave you unprotected.
